integrated_channels/canvas/client.py

```diff
@@ -183,28 +234,33 @@ def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-
         pass
 
     # Private Methods
-
-    def _update_course_details(self, course_id, serialized_data):
+    def _update_course_details(self, course_id, course_details):
         """
         Update a course for image_url (and possibly other settings in future)
-        This is used only for settings that are not settable in the initial course creation
+        Also sets course to 'offer' state by sending 'course[event]=offer'
+
+        Arguments:
+          - course_details (dict): { 'image_url' } : optional used if present to course[image_url]
         """
+        response_code = None
+        response_text = None
+        # Providing the param `event` and setting it to `offer` is equivalent to publishing the course.
+        update_payload = {'course': {'event': 'offer'}}
         try:
             # there is no way to do this in a single request during create
             # https://canvas.instructure.com/doc/api/all_resources.html#method.courses.update
-            content_metadata_item = json.loads(serialized_data.decode('utf-8'))['course']
-            if "image_url" in content_metadata_item:
-                url = CanvasAPIClient.course_update_endpoint(
-                    self.enterprise_configuration.canvas_base_url,
-                    course_id,
-                )
-                # Providing the param `event` and setting it to `offer` is equivalent to publishing the course.
-                self._put(url, json.dumps({
-                    'course': {'image_url': content_metadata_item['image_url'], 'event': 'offer'}
-                }).encode('utf-8'))
+            url = CanvasAPIClient.course_update_endpoint(
```

Comment:
this refactor is here because we always want to set course to offer state, but the image_url detection of optional value is a separate concern, we don't want to mix the two